file_store/src/file_info_poller.rs

```diff
@@ -395,36 +412,58 @@ impl FileInfoPollerState for sqlx::Pool<sqlx::Postgres> {
             .map_err(Error::from)
     }
 
+    async fn oldest_timestamp(
+        &self,
+        process_name: &str,
+        file_type: &str,
+    ) -> Result<Option<DateTime<Utc>>> {
+        sqlx::query_scalar(
+            r#"
+            SELECT file_timestamp FROM files_processed
+            WHERE process_name = $1 AND file_type = $2
+            ORDER BY file_timestamp DESC
+            OFFSET $3
+            LIMIT 1
+            "#,
+        )
+        .bind(process_name)
+        .bind(file_type)
+        .bind(MAX_SIZE_OF_FILES_PROCESSED_TABLE - 1)
+        .fetch_one(self)
```

Comment:
won't this error if there are not 99 entries matching the where clause?

Reply:
Ah I meant to do fetch optional then